src/pages/dataTableUtilities/reducer/reducerMethods.js

```diff
@@ -252,3 +262,124 @@ export const sortData = (state, action) => {
+ * Action: { type: 'closeBasicModal' }
+ */
+export const closeBasicModal = state => ({ ...state, modalIsOpen: false, modalKey: '' });
+
+/**
+ * Adds all items from a master list to the pageObject held
+ * in state - either a Requisition or Transaction.
+ * @param {Object} state  The current state
+ * @param {Object} action The action to act upon
+ * Action: { type: 'addMasterListItems', objectType }
+ */
+export const addMasterListItems = (state, action) => {
+  const { pageObject, database, backingData } = state;
+  const { objectType } = action;
+
+  database.write(() => {
+    pageObject.addItemsFromMasterList(database);
+    database.save(objectType, pageObject);
+  });
+
+  const newData = backingData.slice();
+
+  return { ...state, data: newData };
+};
+
+/**
+ * Creates an Item (Either Requisition or Transaction), and appends
+ * this item to the data array for a page.
+ *
+ * @param {Object} state  The current state
+ * @param {Object} action The action to act upon
+ * Action: { type: 'adadItem', item, addedItemType }
+ */
+export const addItem = (state, action) => {
+  const { database, pageObject, data } = state;
+  const { item, addedItemType } = action;
+  let addedItem;
+
+  database.write(() => {
+    if (pageObject.hasItem(item)) return;
+    addedItem = createRecord(database, addedItemType, pageObject, item);
+  });
+
+  if (addedItem) return { ...state, data: [addedItem, ...data], modalIsOpen: false };
+
+  return { ...state, modalIsOpen: false };
+};
+
+/**
+ * Edits the passed pageObject field with the value supplied.
+ * Used for simple value setting i.e. comments.
+ *
+ * @param {Object} state  The current state
+ * @param {Object} action The action to act upon
+ * Action: { type: 'editPageObject', value, field, pageObjectType }
+ */
+export const editPageObject = (state, action) => {
+  const { database, pageObject } = state;
+  const { value, pageObjectType, field } = action;
+
+  if (pageObject[field] !== value) {
+    database.write(() => {
+      pageObject[field] = value;
+      database.save(pageObjectType, pageObject);
+    });
+  }
+
+  return { ...state, modalIsOpen: false };
+};
+
+/**
+ * Deletes the selected RequisitionItems or
+ * TransactionItems held in state. Where each selected
+ * item is indicated by DataState[rowKey].isSelected.
+ *
+ * @param {Object} state  The current state
+ * @param {Object} action The action to act upon
+ */
+export const deleteItemsById = (state, action) => {
+  const { database, pageObject, dataState, hasSelection, backingData } = state;
+  const { pageObjectType } = action;
+
+  if (!hasSelection) return state;
+
+  const itemsById = Array.from(dataState.keys()).filter(rowKey => dataState.get(rowKey).isSelected);
+
+  database.write(() => {
+    pageObject.removeItemsById(database, itemsById);
+    database.save(pageObjectType, pageObject);
+  });
+
+  const newDataState = new Map();
+  const newData = backingData.slice();
+
+  return {
+    ...state,
+    data: newData,
```

Comment:
While it is a bit janky, I think it's ok so long as 
A: The database.write() is synchronous and runs after the newData slice
B: Isn't too slow (though it's only on deleting so not too problematic if there is some delay)

Reply:
I think it's ok for now. Though some thunkers or something would work well. Possibly have a think about implementing our own simple database middleware, like I think you suggested. Would be an easy migration if we decide to go that route, I think